lib/kernel.js

```diff
@@ -1,20 +1,27 @@
 'use babel';
 
 import { Emitter } from 'atom';
+import { observable } from 'mobx';
+
+import { grammarToLanguage } from './utils';
+import store from './store';
 
-import StatusView from './status-view';
 import WatchSidebar from './watch-sidebar';
 import HydrogenKernel from './plugin-api/hydrogen-kernel';
 import log from './log';
 
 export default class Kernel {
+  @observable executionState = 'loading'
```

Comment:
This is actually everything that's necessary for MobX to do it's magic. 👀 
I really enjoy it for our use case. c5b27c41a4b3f689231a850aba62e3ec1358111b Simplified it a lot.